src/TimeService.Business/Commands/Stat/FindStatCommand.cs

```diff
@@ -137,17 +142,30 @@ public async Task<FindResultResponse<StatInfo>> ExecuteAsync(FindStatFilter filt
         ? await _workTimeMonthLimitRepository.GetAsync(filter.Year, filter.Month.Value)
         : null;
 
-      Task<(List<UserData>, int)> usersTask = _userService.GetFilteredUsersDataAsync(usersIds, filter.SkipCount, filter.TakeCount, filter.AscendingSort, errors);
-      Task<List<CompanyData>> companiesTask = _companyService.GetCompaniesDataAsync(usersIds, errors);
+      (List<UserData> usersData, int totalCount) = await _userService.GetFilteredUsersDataAsync(
+        usersIds,
+        filter.SkipCount,
+        filter.TakeCount,
+        filter.AscendingSort,
+        errors);
 
-      await Task.WhenAll(usersTask, companiesTask);
+      Task<List<CompanyData>> companiesTask = _companyService.GetCompaniesDataAsync(
+        usersData.Select(ud => ud.Id).ToList(),
+        errors);
+      Task<List<ImageData>> imagesTask = _imageService.GetImagesAsync(usersIds, errors);
+      Task<List<PositionData>> positionsTask = _positionService.GetPositionsAsync(usersIds, errors);
 
-      List<CompanyUserData> companies = (await companiesTask)?.SelectMany(p => p.Users).ToList();
+      await Task.WhenAll(
+        companiesTask,
+        imagesTask,
+        positionsTask);
 
-      (List<UserData> usersData, totalCount) = await usersTask;
+      List<CompanyUserData> companies = (await companiesTask)?.SelectMany(p => p.Users).ToList();
 
       List<UserInfo> usersInfos = usersData
-        ?.Select(u => _userInfoMapper.Map(u, companies?.FirstOrDefault(p => p.UserId == u.Id))).ToList();
+        .Select(u => _userInfoMapper.Map(
+          u,
+          imagesTask.Result.FirstOrDefault(i => i.ParentId == u.Id))).ToList();
```

Comment:
use await instead of .result

Reply:
It's not possible in this context or at least I have no idea how to